packages/ui/src/helpers/auth.ts

```diff
@@ -48,12 +50,24 @@ export enum FederatedIdentityProviders {
 
 /**
  * Given xstate context from AuthMachine, this returns the input label, type,
- * and error attributes of the configured login_mechanisms.
+ * and error attributes of the configured login_mechanisms. An optional "alias"
+ * may be passed in to get info from context for that specific alias.
  */
-export const getAliasInfoFromContext = (context: AuthContext) => {
+export const getAliasInfoFromContext = (
+  context: AuthContext,
+  alias?: LoginMechanism
+) => {
   const loginMechanisms = context.config?.login_mechanisms ?? ['username'];
   const error = context.actorRef?.context?.validationError['username'];
 
+  if (alias as UserNameAlias) {
+    return {
+      label: authInputAttributes[alias].label,
+      type: authInputAttributes[alias].type,
+      error,
+    };
+  }
```

Comment:
Same here. I don't believe type casting can be used in place of filtering.

https://www.typescriptlang.org/play?#code/C4TwDgpgBAggNgSwIYGcoF4oHICuKIBOAdkgLYRZQA+2EpSCclNWYAFgPZEUDcAsAChQkKAGUOAY2RwM2AGZIJEAEYcOAa2bYA5mu1wK1bGSQAvLln5Dw0AFIcERWfGRoa4qUjhXBErimAoJAICAC4oe0cAbQBdWSjcfGIyCgAaWnpGLHSsE3MibOx2LjSdPQMsGKs-IgCoOUZgQggAE1lgggA6BrgmggAKfoA3LwBKDAA+KBGZVFhEVFGfARqUDgNOuA5tfqwevtaoBDQAWhPsJd9-dYhN7f795pbLgUE5DgIoftXAmagOORBELjADegigR0Bwy8QTQLkWUDBAghENWNzuOxmLwhAF9BDigA